test/ForkableRealityETH.t.sol

```diff
@@ -0,0 +1,1145 @@
+            "Child2", // string.concat(IERC20Metadata(address(_forkonomicToken)).name(), "1"),
+            "Child2" // IERC20Metadata(address(_forkonomicToken)).symbol()
+        );
+
+        return (forkonomicToken1, forkonomicToken2);
+    }
+
+    function _forkRealityETH(
+        address _forkableRealityETH,
+        address _forkonomicToken1,
+        address _forkonomicToken2,
+        bytes32 _forkOverQuestionId
+    ) internal returns (address, address) {
+        address parentForkmanager = ForkableRealityETH_ERC20(
+            _forkableRealityETH
+        ).forkmanager();
+
+        vm.prank(parentForkmanager);
+        (
+            address forkableRealityETH1,
+            address forkableRealityETH2
+        ) = ForkableRealityETH_ERC20(_forkableRealityETH).createChildren();
+
+        vm.prank(parentForkmanager);
+        ForkableRealityETH_ERC20(forkableRealityETH1).initialize(
+            ForkonomicToken(_forkonomicToken1).forkmanager(),
+            _forkableRealityETH,
+            _forkonomicToken1,
+            _forkOverQuestionId
+        );
+        vm.prank(parentForkmanager);
+        ForkableRealityETH_ERC20(forkableRealityETH2).initialize(
+            ForkonomicToken(_forkonomicToken2).forkmanager(),
+            _forkableRealityETH,
+            _forkonomicToken2,
+            _forkOverQuestionId
+        );
+
+        vm.prank(parentForkmanager);
+        ForkableRealityETH_ERC20(_forkableRealityETH).handleInitiateFork();
+
+        ForkableRealityETH_ERC20(_forkableRealityETH).handleExecuteFork();
+
+        return (forkableRealityETH1, forkableRealityETH2);
+    }
+
+    function _testInitialize(address _forkonomicToken) internal {
+        assertEq(
+            ForkableRealityETH_ERC20(forkableRealityETH).forkmanager(),
+            forkmanager
+        );
+        assertEq(
+            ForkableRealityETH_ERC20(forkableRealityETH).parentContract(),
+            address(0)
+        );
+
+        vm.expectRevert("Initializable: contract is already initialized");
+        // vm.expectRevert(IForkableStructure.NotInitializing.selector); In future will be this
+        ForkableRealityETH_ERC20(forkableRealityETH).initialize(
+            forkmanager2,
+            forkableRealityETH,
+            _forkonomicToken,
+            bytes32(0)
+        );
+    }
+
+    function testInitialize() public {
+        _testInitialize(forkonomicToken);
+    }
+
+    function testClaim() public {
+        _doClaim(forkableRealityETH, importFinalizedClaimedQuestionId);
+    }
+
+    function testHandleForkOnlyAfterForking() public {
+        // Testing revert if children are not yet created
+        vm.prank(forkmanager);
+        ForkableRealityETH_ERC20(forkableRealityETH).handleInitiateFork();
+        vm.expectRevert(IForkableStructure.OnlyAfterForking.selector);
+        ForkableRealityETH_ERC20(forkableRealityETH).handleExecuteFork();
+    }
+
+    function _testTemplateCreation(address _forkableRealityETH) internal {
```

Comment:
Hmm, I think _test is clearer. Functionally it's doing just the same as it would if it was a test function except that it takes a parameter so we can run it on a few different versions, that's why it's an internal function called by an external one.